### ISSUE TITLE
[E6E] Atum worldgen hangs "server" during singleplayer session

### ISSUE THREAD

**Issue by zippedfreak** [EnigmaticaModpacks/Enigmatica6#4598]
### Modpack Version

1.0 and 1.1.2

### Describe your issue.

I've played this pack (enigmatica 6 expert) with barely any issues for a while now until i hit the point in progression where i need to enter the atum dimension. As i enter the dimension everything is working fine for a few seconds until the worldgen encounters an issue and freezes the world in place. I'm free to move around and the game keeps running but the "server" is not responding. I have a clean save that i have been using to keep trying different things to fix the issue. I even managed to save the game in the frozen state that i can load into but it doesn't fix the issue. 

Things i have so far tried in my troubleshooting:
- Changing game version
- Allocating different amounts of ram from 8gb-17gb
- switched from curgeforge launcher to multimc
- destroying portal and placing it in a different spot but that does not seem to either change the seed nor the entrance location in atum
- Used the Pregen mod to preload the dimension (as suggested in another thread) but that also get's stuck (basically the same outcome but i'm instead standing the the overworld) when trying to expand the load from 100 to 200 chunks
- Used a tool called NBTexplorer and changed the worldgen generator and biome source seeds in the level.dat file, deleted the Atum folder but ended up with the same Atum world with the same issue anyways. Unsure if the change get's overridden or if my changes are not saved properly. 

I've posted the latest.log file and this seems to the the line where the issue occurs. 

[08Apr2022 09:40:06.235] [Server thread/WARN] [net.minecraft.server.MinecraftServer/]: Can't keep up! Is the server overloaded? Running 2224ms or 44 ticks behind

### Latest Log

https://gist.github.com/zippedfreak/e8f68beb5bc1c5aa8aec95571e6e9311

### Have you modified the modpack?

No

### Did the issue happen in singleplayer or on a server?

Singleplayer

**Comment by zippedfreak** [comment]
I have also now entered Atum in a new save after spawning in the requirements via cheats and there seems to be no issues after flying around for a few minutes.

**Comment by zippedfreak** [comment]
> 

Thank you! 
As a bit of an update i managed to enter the atum dimension without a server stall after lowering my render distance to 4 and running around in a circle to generate terrain at a slower rate. I managed to explore a larger zone than i previously have so i went back to the overworld and tried to pregen 200 chunks to test if it was fixed. The pregen seemed to have worked so i tried entering the dimension again and the game crashed. After starting the game up again it seems to be finally working as i've managed to keep exploring for a while without any hiccups so far. I'll keep testing later today but hopefully it's resolved itself permanently.

I included the logs for the crash if that is of any value. 
Latest.log https://gist.github.com/zippedfreak/1275b580a3c7a5feeafa7f1d8eedaa4e
debug.log https://gist.github.com/zippedfreak/6f79a906a511b1d18f838b50fc85116f
Crash report https://gist.github.com/zippedfreak/6f79a906a511b1d18f838b50fc85116f

**Comment by jkirkpatrick** [comment]
Experiencing the same issue on singleplayer - started with E6E 1.0 (no problem in Atum), problem started after upgrad to 1.1.2. Dimension seems to be stuck / "Waiting for server..."

**Comment by zippedfreak** [comment]
> Experiencing the same issue on singleplayer - started with E6E 1.0 (no problem in Atum), problem started after upgraded to 1.1.2. Dimension seems to be stuck / "Waiting for server..."
> 
> https://gist.github.com/jkirkpatrick/09153004d032a534423e4341ddb0fe83

So you already had a lot of Atum already generated before you encountered the issue? Have you tried lowering render distance or does the issue occur as you enter the world regardless?

**Comment by jkirkpatrick** [comment]
> 

Yes, but not a lot of chunks generated, maybe 100 (or whatever the default area generation around the portal). I had not done any exploring - just generated the portal prior to upgrade - and popped to see if it was in a good location. Problem may have been the result of a sandstorm. GFX settings were already on low as it related to particles, but render distance was default. After a restart and reload I could not reproduce the issue - the sandstorm was over.

**Comment by tyler489** [comment]
We have a crash at a specific chunk on the server, seems to be relating to a pillager encampment as the player get the acheivement "**Who's the Pillager Now?. Give a Pillager a taste of their own medicine.** before the server hangs, nothing ticks, nothing moves, even chat stops untill the server is restarted.
